src/python/pants/backend/typescript/target_types.py

```diff
@@ -57,7 +58,9 @@ class TypeScriptSourcesOverridesField(OverridesField):
 
 
 class TypeScriptSourcesGeneratorSourcesField(TypeScriptGeneratorSourcesField):
-    default = tuple(f"*{ext}" for ext in TS_FILE_EXTENSIONS)
+    default = tuple(f"*{ext}" for ext in TS_FILE_EXTENSIONS) + tuple(
+        f"!{pat}" for pat in TS_TEST_FILE_EXTENSIONS
+    )
```

Comment:
This is necessary to do to avoid having source modules owning test modules. This is how it's done for Python sources, for example:

https://github.com/pantsbuild/pants/blob/255db66324c5c0a9184ad261ff1c793bbaf0c107/src/python/pants/backend/python/target_types.py#L1069-L1074

Reply:
Excellent! Could JS also get this update?

Reply:
Yes, in the next PR.